pkg/utils/v1alpha1/kubernetes.go

```diff
@@ -173,7 +190,13 @@ func FetchAndUpdateVolInfos(nodeID string) (err error) {
 	}
 
 	for _, csivol := range csivols.Items {
+		if csivol.DeletionTimestamp != nil {
+			continue
+		}
 		vol := csivol
+		if vol.Status == apis.CSIVolumeStatusMountUnderProgress {
+			vol.Status = apis.CSIVolumeStatusUninitialized
+		}
```

Comment:
It would have been more readable if we tried predicates to filter from list

Reply:
This will also be taken up in a separate PR as the predicates are not being used anywhere in the repo, this can be done during the refactoring.